orchagent/bufferorch.cpp

```diff
@@ -579,7 +579,19 @@ task_process_status BufferOrch::processQueue(Consumer &consumer)
     }
     else
     {
-        SWSS_LOG_ERROR("Queue profile '%s' was inserted after BufferOrch init", key.c_str());
+        // If a buffer queue profile is not in the initial CONFIG_DB BUFFER_QUEUE table
+        // at BufferOrch object instantiation, it is considered being applied
+        // at run time, and, in this case, is not tracked in the m_ready_list. It is up to
+        // the application to guarantee the set order that the buffer queue profile
+        // should be applied to a physical port before the physical port is brought up to
+        // carry traffic. Here, we alert to application through syslog when such a wrong
+        // set order is detected.
+        for (const auto &port_name : port_names)
+        {
+            if (gPortsOrch->isPortAdminUp(port_name)) {
```

Comment:
When PortOrch receives port-up, can the PortOrch query bufferOrch, if port is allotted with buffers or not ? If not, it could fire a self timer and delay the processing of the port UP, until appropriate buffers are allotted. Each time the process is deferred, it could post a WARNING LOG.